backend/dataall/modules/dataset_sharing/aws/glue_client.py

```diff
@@ -130,3 +132,38 @@ def delete_database(self):
                 f'due to: {e}'
             )
             raise e
+
+    def get_source_catalog(self):
+        """ Get the source catalog account details """
+        try:
+            log.info(f'Fetching source catalog details for database {self._database}...')
+            response = self._client.get_database(CatalogId=self._account_id, Name=self._database)
+            linked_database = response.get('Database', {}).get('TargetDatabase', {})
+            log.info(f'Fetched source catalog details for database {self._database} are: {linked_database}...')
+            if linked_database:
+                return Catalog(account_id=linked_database.get('CatalogId'),
+                               database_name=linked_database.get('DatabaseName'),
+                               region=linked_database.get('Region', self._region))
+        except Exception as e:
```

Comment:
This is OK but also a bit of a smell to wrap entire method in try/catch with generic Exception.. It would be better to catch specific exceptions.

Reply:
Removed generic exception and added a specifc exception ( EntityNotFoundException ). Although entire try catch could be used removed, using this specific exception for more informative logging